### PR TITLE
Custom root certificates for JWKSResolver

### LINKED ISSUE #15086: Allow for custom RootCA certs when resolving JWKS
**Describe the feature request**
When Pilot resolves a JWKS URI, it should be configurable which Root CAs it trusts; currently, it is hard-coded to `/cacert.pem` (coming from `./tests/testdata/certs/cacert.pem`), which are only public CAs, obviously. If a customer uses a corporate CA for the server hosting the JWKS, Pilot will fail to resolve the URI.

Pilot should offer to configure the `publicRootCABundlePath` which is currently hard-coded to `/cacert.pem` via an environment variable, CLI parameter or the MeshConfig. This way, we could mount a fitting certificate collection into the Pilot container.

**Describe alternatives you've considered**
- proxying through a HTTP server (breaks zero-trust)
- overwriting `/cacert.pem` in Pilot's container in a separate layer (should not be necessary for standard use cases IMO)

**Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[x] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure

**Additional context**

### PR DESCRIPTION
**Pull request by dgn**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Fixes #15086, should also work for #9505

This PR adds a variable to the `pilot` helm chart to inject a trusted CA into the JWKSResolver. This can be necessary when hosting your JWKS on a server protected with a certificate from an internal CA.

When installing via `helm`, you can now specify a PEM-encoded certificate in `pilot.jwksResolverExtraRootCA` in your `values.yaml` to trust an additional CA certificate:
```yaml
...
pilot:
  enabled: true
  jwksResolverExtraRootCA: |
    -----BEGIN CERTIFICATE-----
    [...]
    -----END CERTIFICATE-----
...
```

Or, during installation you can directly specify your certificate as a command-line argument:

```bash
helm install [...] --set-file pilot.jwksResolverExtraRootCA=yourRootCertificate.pem
```

### CODE REVIEW

File: install/kubernetes/helm/istio/charts/pilot/templates/deployment.yaml
Comment (rshriram) [inline install/kubernetes/helm/istio/charts/pilot/templates/deployment.yaml]: why is this extra? shouldnt it be istio-jwks-root-ca ?

Comment (dgn) [reply install/kubernetes/helm/istio/charts/pilot/templates/deployment.yaml]: Well, we include a bundle of standard CA certificates in the container, anything you add here will *also* be trusted, in addition to that included bundle.

**Comment by pitlv2109** [comment]
@geeknoid @howardjohn This is tagged as `area/security` (indeed it is), can we do something so that bot will request code review from the security group?

**Comment by howardjohn** [comment]
> This is tagged as area/security (indeed it is), can we do something so that bot will request code review from the security group?

That would be nice. I suspect its pretty infrequent to have a security PR that doesn't require security owners approval though

File: pilot/pkg/model/test/testcert/generate.sh
Comment (howardjohn) [inline pilot/pkg/model/test/testcert/generate.sh]: thank you for adding this

**Review by howardjohn — COMMENTED** [review]
lgtm, will wait for security folks to approve.

@yangminzhu

**Review by yangminzhu — APPROVED** [review]
Thanks for the PR.
Could you add instructions of how to use the extra CA bundle in the PR description and also make sure to update the 1.4 release notes for this?

**Comment by yangminzhu** [comment]
@howardjohn Could you approve if this looks good to you? Thanks!

**Review by howardjohn — APPROVED** [review]
Please send a PR to istio/install so this can be used from the operator

**Pull request merged**
